constraint/pkg/externaldata/cache.go

```diff
@@ -69,6 +76,53 @@ func isValidURL(url string) bool {
 	return true
 }
 
+func isValidCABundle(provider *v1alpha1.Provider) error {
+	verify := func(caBundle string) error {
+		caPem, err := base64.StdEncoding.DecodeString(caBundle)
+		if err != nil {
+			return fmt.Errorf("failed to base64 decode caBundle: %w", err)
+		}
+
+		caDer, _ := pem.Decode(caPem)
+		if caDer == nil {
+			return fmt.Errorf("bad caBundle")
+		}
+
+		_, err = x509.ParseCertificate(caDer.Bytes)
+		if err != nil {
```

Comment:
Add test that ensures we fail as expected when caDer.Bytes fails to parse.

Also - add comment on why we're doing this parsing. Since it isn't obvious _why_ we're doing this, it would be easy for a future code editor to remove this check. No tests would fail and there's no comment explaining why this check is necessary.

Reply:
> Add test that ensures we fail as expected when caDer.Bytes fails to parse.

Do you have an example certificate for this test case?

Reply:
Hmm most of the bad CA bundles I've tried were caught by `pem.Decode`. Still trying to look for a certificate that can pass `pem.Decode` but not `x509.ParseCertificate`